### PR TITLE
VIITE-767 changes to project link partitioning logic

### PR DESCRIPTION
**Pull request by tvenhola**
grouping of links by being in project or not and subgroup rules
applied the 0.1 meter connectedness rule

### CODE REVIEW

**Comment by oappicgi** [comment]
http://localhost:9001/api/viite/project/roadlinks?zoom=12&bbox=546243.9394773282,6759817.977519371,547208.9394773282,6760886.977519371&id=7091389 still results nullpointer exception for me in vvhclient 622

**Comment by tvenhola** [comment]
This presumably happens because some of the suravage links are duplicated in VVH.

File: digiroad2-geo/src/main/scala/fi/liikennevirasto/digiroad2/linearasset/GraphPartitioner.scala [hunk 1]
```diff
@@ -8,8 +8,8 @@ import org.geotools.graph.structure.basic.BasicEdge
 import scala.collection.JavaConversions._
 
 trait GraphPartitioner {
-  protected def clusterLinks[T <: PolyLine](links: Seq[T]): Seq[Graph] = {
-    val generator = new BasicLineGraphGenerator(0.5)
+  protected def clusterLinks[T <: PolyLine](links: Seq[T], tolerance: Double = 0.5): Seq[Graph] = {
```
Comment (Campanha) [inline digiroad2-geo/src/main/scala/fi/liikennevirasto/digiroad2/linearasset/GraphPartitioner.scala]: should we use fi.liikennevirasto.viite.MaxSuravageToleranceToGeometry for tolerance value?

Comment (tvenhola) [reply digiroad2-geo/src/main/scala/fi/liikennevirasto/digiroad2/linearasset/GraphPartitioner.scala]: Partitioner can use any given value, but for viite the "connectedness" is defined as < 0.1 meters. (It's used in projectlinkpartitioner, for example)

**Pull request merged**